governance/proposals/010-arbitrum-l1-l2-messaging.js

```diff
@@ -0,0 +1,239 @@
+        type: 'address',
+      },
+      {
+        internalType: 'uint256',
+        name: 'gasLimit',
+        type: 'uint256',
+      },
+      {
+        internalType: 'uint256',
+        name: 'maxFeePerGas',
+        type: 'uint256',
+      },
+      {
+        internalType: 'bytes',
+        name: 'data',
+        type: 'bytes',
+      },
+    ],
+    name: 'createRetryableTicket',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256',
+      },
+    ],
+    stateMutability: 'payable',
+    type: 'function',
+  },
+]
+
+/**
+ * Set up: instantiate L1 / L2 wallets connected to providers
+ */
+const walletPrivateKey = process.env.PRIVATE_KEY
+const L1RPC = 'https://rpc.unlock-protocol.com/1' // mainnet RPC
+const L2RPC = 'https://rpc.unlock-protocol.com/42161' // Arbitrum RPC
+const l1Provider = new ethers.JsonRpcProvider(L1RPC)
+const l2Provider = new ethers.JsonRpcProvider(L2RPC)
+// const l1Wallet = new ethers.Wallet(walletPrivateKey, l1Provider)
+const l2Wallet = new ethers.Wallet(walletPrivateKey, l2Provider)
+
+module.exports = async () => {
+  console.log('Cross-chain Proposer')
+
+  const l2Network = await getL2Network(l2Provider)
+  const ethBridger = new EthBridger(l2Network)
+  const inboxAddress = ethBridger.l2Network.ethBridge.inbox
+
+  const L2TokenContract = new ethers.Contract(
+    ARB_TOKEN_ADRESS_ON_L2,
+    ERC20_ABI,
+    l2Wallet
+  ).connect(l2Wallet)
+
+  const balanceOf = await L2TokenContract.balanceOf(TIMELOCK_L2_ALIAS)
+  const tokenAmount = ethers.parseEther('1')
+
+  // Create an instance of the Interface from the ABIs
+  const iface_erc20 = new ethers.Interface(ERC20_ABI)
+  const iface_inbox = new ethers.Interface(INBOX_ABI)
+
+  // Encode the ERC20 Token transfer calldata
+  const transfer_calldata = iface_erc20.encodeFunctionData('transfer', [
+    GRANTS_CONTRACT_ADDRESS,
+    tokenAmount,
+  ])
+  /**
+   * Now we can query the required gas params using the estimateAll method in Arbitrum SDK
+   */
+  const l1ToL2MessageGasEstimate = new L1ToL2MessageGasEstimator(l2Provider)
+
+  /**
+   * The estimateAll method gives us the following values for sending an L1->L2 message
+   * (1) maxSubmissionCost: The maximum cost to be paid for submitting the transaction
+   * (2) gasLimit: The L2 gas limit
+   * (3) deposit: The total amount to deposit on L1 to cover L2 gas and L2 call value
+   */
+  const L1ToL2MessageGasParams = await l1ToL2MessageGasEstimate.estimateAll(
+    {
+      from: L1_TIMELOCK_CONTRACT,
+      to: ARB_TOKEN_ADRESS_ON_L2,
+      l2CallValue: 0,
+      excessFeeRefundAddress: L1_TIMELOCK_CONTRACT,
+      callValueRefundAddress: L1_TIMELOCK_CONTRACT,
+      data: transfer_calldata,
+    },
+    await getBaseFee(l1Provider),
+    l1Provider
+  )
+  const gasPriceBid = await l2Provider.getGasPrice()
+  const ETHDeposit = L1ToL2MessageGasParams.deposit.toNumber() * 10 // I Multiply by 10 to add extra in case gas changes due to proposal delay
```

Comment:
ETHDeposit is the amount ETH value that must be sent to the Delayed Inbox Contract for a successful retryable ticket creation. It is estimated by calling the ```l1ToL2MessageGasEstimate.estimateAll()``` function provided by the Arbitrum sdk. When the ```createRetryableTicket`` function is called, it forces the caller to pay this amount else reverts, this is why the ```excessFeeRefundAddress``` you asked about earlier is needed, to send any excess back. Here I multiply the actual value by 10 as a buffer because in the time it takes to execute a proposal that value may change.